### ISSUE TITLE
Error With Certain Expectations When Using Trino (SqlAlchemyExecutionEngine) Or SparkDF (SparkDFExecutionEngine) Connected to Delta Lake Table

### ISSUE THREAD

**Issue by rylativity** [fivetran/great_expectations#7206]
**Describe the bug**
I have a Trino instance that is connected to and can query a Delta Lake table.  I have added the Trino database as a datasource in a Great Expectations context using the SQLAlchemy Execution Engine as instructed [here](https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/database/trino/). 

I am able to run metadata-level expectations (e.g. expect_column_to_exist) without any error, and they run as expected. 

However, I get an error when attempting to run any data-level expectations (e.g. `validator.expect_column_min_to_be_between(col, min, max)`) or if I call `validator.head()`.

I also attempted to use the SparkDF Execution Engine as instructed [here](https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/in_memory/spark/) to create a SparkDF from the Delta Lake table, and saw similar behavior - metadata-level expectations run as expected, while data-level expections or `validator.head()` calls failed.

I have saved rendered copies of rendered Jupyter notebooks showing the error stacktraces for both Trino and Spark (errors are at the bottom of each notebook; successful metadata-level expectations are in the cells just above the errors).

- Spark Error Example: https://github.com/rylativity/container-analytics-platform/blob/great_expectations/jupyter/notebooks/great_expectations_delta_example.ipynb
- Trino Error Example: https://github.com/rylativity/container-analytics-platform/blob/great_expectations/jupyter/notebooks/great_expectations_trino_example.ipynb

**To Reproduce**
1. Create a Delta Lake Table
2. Run a Trino instance, add your Delta Lake configuration to the Trino catalog, and register the Delta Lake table in Trino
3. Follow the [Great Expectation's Trino connection documentation](https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/database/trino/) to connect to your Trino instance
4. Run data-level expectations (e.g. `validator.expect_column_min_to_be_between(col, min, max)`) on your datasource and see error. 

[This repository](https://github.com/rylativity/container-analytics-platform/tree/great_expectations), which is the same repository holding the rendered notebooks with errors linked above, can also be used to recreate the environment using docker-compose.

**Expected behavior**
Expect data-level expectations (e.g. `validator.expect_column_min_to_be_between()`) and `validator.head()` calls to run without error.

**Environment (please complete the following information):**
 - Docker running on Ubuntu 22.04 host (services defined in this [docker-compose.yml](https://github.com/rylativity/container-analytics-platform/blob/great_expectations/docker-compose.yml)
 - Great Expectations Version: 0.15.49

**Additional context**
Reading the Delta Lake table into memory, converting it to a Pandas dataframe, and using the Pandas Execution Engine works as expected (without error) for both data-level and metadata-level expectations.

**Comment by alexsherstinsky** [comment]
Hello, @rylativity -- thank you very much for submitting this issue with a wonderfully detailed report and the notebooks.  I took a look at the `Spark` notebook and have an idea about what is happening; however, it seems to me that the actual exception trace that we need to better understand the root cause of the problem is, while similar in spirit to the one appearing in the notebook, should actually be different.  Hence, if there is interest, please reach out to me in the Great Expectations Slack Community so that we can set up a time to troubleshoot it together via screen sharing.  (This might be faster than having us reproduce the Trino data lake environment; however, if doing it live is not an option, please let us know, and we will prioritize this within our team.). Thank you very much again.

**Comment by alexsherstinsky** [comment]
Quick update: @rylativity and I were able to reproduce the issue with Spark (SQL works fine now).  @rylativity was kind enough to share his container environment with me and pair with me so as to reproduce the issue in my environment as well.  Hence, I have the action item to continue troubleshooting.  I will keep everybody up to date.  Thank you!

**Comment by alexsherstinsky** [comment]
@rylativity I believe that I determined the cause by running your container in my environment using a troubleshooting-focused GreatExpectations branch.  More updates coming soon.  Thank you.

**Comment by alexsherstinsky** [comment]
@rylativity I am really pleased to update you that the root of the issue is indeed what we suspected, and a fix I tried worked beautifully.  My next steps are to propose this fix to the team.  It may take a few days, because I would like to simplify the way we handle Spark Session, which would be a conceptual change from the way we have been doing it since the GreatExpectations early days.  I will keep you posted regarding the progress on our side in the coming days.  Thanks!

**Comment by alexsherstinsky** [comment]
@rylativity I am very pleased to inform you that PR #7444 contains the fix to the issue, which you found (and I verified the fix using your container!).  Once it is merged into `develop`, it will be released next Thursday, March 30, 2023.  Thank you!

**Comment by rylativity** [comment]
Awesome! Thanks for getting this fix pushed through, and really happy I was able to be of some assistance.

**Comment by alexsherstinsky** [comment]
The fix has been merged into the `develop` branch and should be part of the release on March 30, 2023.  Thank you!

**Comment by rylativity** [comment]
Confirmed that the 0.16.4 release resolves the issue. Really appreciate the speedy resolution. Thanks, @alexsherstinsky!
